pkg/action/scan_error.go

```diff
@@ -0,0 +1,67 @@
+package action
+
+import "fmt"
+
+const (
+	errMsgScanFailed     = "scan failed"
+	errMsgGenerateFailed = "failed to generate file report"
+	errMsgNilReport      = "file report is nil"
+)
+
+type ErrorType int
+
+const (
+	TypeScanError ErrorType = iota
+	TypeGenerateError
+	TypeNilError
+)
+
+type FileReportError struct {
+	err    error
+	path   string
+	reason string
+}
+
+func NewFileReportError(err error, path, reason string) *FileReportError {
```

Comment:
```suggestion
func NewFileReportError(err error, path, reason ErrorType) *FileReportError {
```

Using a freeform string here makes it harder for external packages to use the typing effectively since they need to be exact string matches for the type switch to work. Consider using the ErrorTypes enum directly here, and converting them to strings in Error().

Reply:
If you do this though, add a new TypeUnknown enum so that 0 defaults to Unknown rather than TypeScanError

Reply:
Added in [`8e1e5c6` (#777)](https://github.com/chainguard-dev/malcontent/pull/777/commits/8e1e5c6cf378a547f46bccdd7d73ceb9a912eb5b).